orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -162,9 +162,10 @@ def build_gradient_accumulation_graph(grad_model, all_args_requiring_gradient_na
         grad_accumulation_buffer_input.name = grad_accumulation_buffer_name
         graph_inputs.append(grad_accumulation_buffer_input)
 
-        # accumulated gradient is also a graph output
-        grad_accumulation_output = copy.deepcopy(graph_output)
-        grad_accumulation_output.name = grad_accumulation_output_name
+        # accumulated gradient update flag is also a graph output
+        grad_accumulation_output = onnx.helper.make_tensor_value_info(
+            grad_accumulation_output_name, onnx.TensorProto.BOOL, [1]
+            )
```

Comment:
@askhade had a TODO in her previous [CAPI PR](https://github.com/microsoft/onnxruntime/pull/11758/files#diff-c436ee39acf75b17c29c0c06ff46a6cbe13577695964a5e49d83b1e78343c837R311). Now that we no longer have the tensors being output, but a bunch of bools, who will be responsible for releasing that memory?
Maybe the ortvalues are cleared internally by ORT? If they are, can we remove that TODO?

Reply:
Following the change of not fetching the bools at all, the release logic is simplified